### PR TITLE
Fix getEntry 429 response

### PR DESCRIPTION
**Pull request by m-cat**
The registry had a bug in the transformation to handle uint64 numbers properly. It did not take into account http status codes.

```ts
// Transform the response to add quotes, since uint64 cannot be accurately
// read by JS so the revision needs to be parsed as a string.
transformResponse: function (data) {
```
It throws `Full error: SyntaxError: Unexpected token < in JSON at position 0`
if it receives the following data: 
```html
<head><title>429 Too Many Requests</title></head>
<body>
<center><h1>429 Too Many Requests</h1></center>
<hr><center>openresty/1.19.3.1</center>
</body>
</html>
```

### CODE REVIEW

File: src/registry.ts [hunk 1]
```diff
@@ -337,3 +315,40 @@ export function validateRegistryEntry(name: string, value: unknown, valueKind: s
...
 }
+
+/**
+ * Handles error responses returned in getEntry.
+ *
+ * @param err - The Axios error.
+ * @returns - An empty signed registry entry if the status code is 404.
+ * @throws - Will throw if the status code is not 404.
+ */
+function handleGetEntryErrResponse(err: AxiosError): SignedRegistryEntry {
```
Comment (peterjan) [inline src/registry.ts:326]: I think we want to throw a custom error here that signals to the SDK user he's being ratelimited.

File: src/registry.ts [hunk 2]
```diff
@@ -337,3 +315,40 @@ export function validateRegistryEntry(name: string, value: unknown, valueKind: s
   validateString(`${name}.data`, (value as RegistryEntry).data, `${valueKind} field`);
   validateBigint(`${name}.revision`, (value as RegistryEntry).revision, `${valueKind} field`);
 }
+
+/**
+ * Handles error responses returned in getEntry.
+ *
+ * @param err - The Axios error.
+ * @returns - An empty signed registry entry if the status code is 404.
+ * @throws - Will throw if the status code is not 404.
```
Comment (peterjan) [inline src/registry.ts:324]: This should be `is 404` right

Comment (peterjan) [inline src/registry.ts:324]: Should also throw imo if status code is 429

File: src/registry.ts (hunk 1 above)
Comment (peterjan) [reply src/registry.ts:326]: It returns this:
```
Error response did not contained expected fields 'data.message'. Status code: 429. Full error: Error: Request failed with status code 429
```

Which is good enough, but I don't think it's the most user friendly error message.

**Review by peterjan — DISMISSED** [review]

File: src/registry.ts [hunk 3]
```diff
@@ -335,6 +335,9 @@ function handleGetEntryErrResponse(err: AxiosError): SignedRegistryEntry {
   // Check if status was 404 "not found" and return null if so.
   if (err.response.status === 404) {
     return { entry: null, signature: null };
+  } else if (err.response.status >= 400) {
+    console.log(err);
```
Comment (peterjan) [inline src/registry.ts]: Why would you `console.log`? Is it a leftover debug line? This shows up in the stdout of anyone importing the SDK and can not be hidden, can get quite spammy.

File: src/registry.ts [hunk 4]
```diff
@@ -335,6 +335,9 @@ function handleGetEntryErrResponse(err: AxiosError): SignedRegistryEntry {
   // Check if status was 404 "not found" and return null if so.
   if (err.response.status === 404) {
     return { entry: null, signature: null };
+  } else if (err.response.status >= 400) {
```
Comment (peterjan) [inline src/registry.ts]: This means that the code following these status code checks are only there to cover `3xx` codes.
I'm just wondering why we can't simply do:
```
if (err.response.status === 404) {
    return { entry: null, signature: null };
}

throw err // rethrow
```
doesn't this accomplish exactly the same. I don't see the value in wrapping the original error the way we do, aside from handling the 404 case.

**Review by peterjan — CHANGES_REQUESTED** [review]
would approve but I think the `console.log` is a debug leftover

File: src/registry.ts (hunk 3 above)
Comment (m-cat) [reply src/registry.ts]: Stop commenting on old commits!! lol

**Review by peterjan — APPROVED** [review]

**Review by dghelm — APPROVED** [review]

**Pull request merged**